src/libraries/System.Private.CoreLib/src/System/Threading/PeriodicTimer.cs

```diff
@@ -22,19 +22,57 @@ public sealed class PeriodicTimer : IDisposable
         private readonly State _state;
 
         /// <summary>Initializes the timer.</summary>
-        /// <param name="period">The time interval between invocations of callback..</param>
+        /// <param name="period">The period between ticks</param>
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="period"/> must represent a number of milliseconds equal to or larger than 1, and smaller than <see cref="uint.MaxValue"/>.</exception>
         public PeriodicTimer(TimeSpan period)
         {
-            long ms = (long)period.TotalMilliseconds;
-            if (ms < 1 || ms > Timer.MaxSupportedTimeout)
+            if (!TryGetMilliseconds(period, out uint ms))
             {
                 GC.SuppressFinalize(this);
                 throw new ArgumentOutOfRangeException(nameof(period));
             }
 
             _state = new State();
-            _timer = new TimerQueueTimer(s => ((State)s!).Signal(), _state, (uint)ms, (uint)ms, flowExecutionContext: false);
+            _timer = new TimerQueueTimer(s => ((State)s!).Signal(), _state, ms, ms, flowExecutionContext: false);
+        }
+
+        /// <summary>Gets or sets the period between ticks.</summary>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> must represent a number of milliseconds equal to or larger than 1, and smaller than <see cref="uint.MaxValue"/>.</exception>
+        /// <remarks>
+        /// All prior ticks of the timer, including any that may be waiting to be consumed by <see cref="WaitForNextTickAsync"/>,
+        /// are unaffected by changes to <see cref="Period"/>. Setting <see cref="Period"/> affects only and all subsequent times
+        /// at which the timer will tick.
+        /// </remarks>
+        public TimeSpan Period
+        {
+            get => TimeSpan.FromMilliseconds(_timer._period);
+            set
+            {
+                if (!TryGetMilliseconds(value, out uint ms))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
```

Comment:
Should this be a `throw helper` to allow `Period` to be inlined or are we thinking its not going to be called frequently enough to matter?

Reply:
I was thinking it'd be called infrequently enough to not matter.